packages/shared-ux/chrome/navigation/src/ui/components/navigation.tsx

```diff
@@ -97,18 +105,40 @@ export function Navigation({ children, unstyled = false, dataTestSubj }: Props)
       register,
       updateFooterChildren: setFooterChildren,
       unstyled,
+      activeNodes,
     }),
-    [register, unstyled]
+    [register, unstyled, activeNodes]
   );
 
   useEffect(() => {
+    const navigationTree = Object.values(navigationItems).sort((a, b) => {
+      const aOrder = orderChildrenRef.current[a.id];
+      const bOrder = orderChildrenRef.current[b.id];
+      return aOrder - bOrder;
+    });
+
+    // We only want to notify the Chrome service if the navigation tree has changed
```

Comment:
Is there a way to call `onProjectNavigationChange` once during the "mounting phase"?  Maybe by debouncing with setTimeout(() =>0, 0) or by tracking children rendering. 

This doesn't look like a performance optimization at this point, but a legit bug with an intermediate not stable state. I assume that my other comment around `// We only want to set the collapsed state during the "mounting" phase (500ms).` is related

Reply:
I will look into it. But IMO `setTimeout(() =>0, 0)` are a hack around the react lifcycle. Maybe its `0ms` maybe we need `100ms`? 

Maybe `useLayoutEffect` could help. But again, I'd like to see the issue we try to fix if there is an issue. 

The comment you mention is not related to this but how the data flow is structured as explained above.

Reply:
With https://github.com/elastic/kibana/pull/159906/commits/e19913a1db39295f476e7b222083e4c92b8b99b9 it has been removed.